src/Analyser/AnalyserResultFinalizer.php

```diff
@@ -92,6 +93,16 @@ public function finalize(AnalyserResult $analyserResult, bool $onlyFiles, bool $
 			}
 		}
 
+		$tempCollectorErrors = array_filter($tempCollectorErrors, function (string $error) use ($scope, $node) : bool {
```

Comment:
I think we should call these extensions right where the error is reported before it's even added to  `$tempCollectorErrors`. Also, don't call the extension for errors that are not ignorable.

Reply:
Applied the feedback